pvlib/iotools/meteonorm.py

```diff
@@ -0,0 +1,723 @@
+        Tilt angle from horizontal plane.
+    surface_azimuth : float, default : 180
+        Orientation (azimuth angle) of the (fixed) plane. Clockwise from north
+        (north=0, east=90, south=180, west=270).
+    time_step : {'1min', '1h'}, default : '1h'
+        Frequency of the time series.
+    horizon : str, optional
+        Specification of the horizon line. Can be either 'flat' or 'auto', or
+        specified as a list of 360 integer horizon elevation angles.
+        'auto'.
+    terrain_situation : str, default : 'open'
+        Local terrain situation. Must be one of: ['open', 'depression',
+        'cold_air_lake', 'sea_lake', 'city', 'slope_south',
+        'slope_west_east'].
+    albedo : float, optional
+        Constant ground albedo. If no value is specified a baseline albedo of
+        0.2 is used and albedo changes due to snow fall are modeled. If a value
+        is specified, then snow fall is not modeled.
+    turbidity : list or 'auto', optional
+        List of 12 monthly mean atmospheric Linke turbidity values. The default
+        is 'auto'.
+    random_seed : int, optional
+        Random seed to be used for stochastic processes. Two identical requests
+        with the same random seed will yield identical results.
+    clear_sky_radiation_model : str, default : 'esra'
+        Which clearsky model to use. Must be either `'esra'` or `'solis'`.
+    data_version : str, default : 'latest'
+        Version of Meteonorm climatological data to be used.
+    future_scenario : str, optional
+        Future climate scenario.
+    future_year : int, optional
+        Central year for a 20-year reference period in the future.
+    interval_index : bool, default : False
+        Index is pd.DatetimeIndex when False, and pd.IntervalIndex when True.
+        This is an experimental feature which may be removed without warning.
+    map_variables : bool, default : True
+        When true, renames columns of the Dataframe to pvlib variable names
+        where applicable. See variable :const:`VARIABLE_MAP`.
+    url : str, optional.
+        Base URL of the Meteonorm API. `'climate/tmy'` is
+        appended to the URL. The default is:
+        :const:`pvlib.iotools.meteonorm.URL`.
+
+    Raises
+    ------
+    requests.HTTPError
+        Raises an error when an incorrect request is made.
+
+    Returns
+    -------
+    data : pd.DataFrame
+        Time series data. The index corresponds to the start (left) of the
+        interval unless ``interval_index`` is set to True.
+    meta : dict
+        Metadata.
+
+    See Also
+    --------
+    pvlib.iotools.get_meteonorm_observation,
+    pvlib.iotools.get_meteonorm_forecast
+
+    References
+    ----------
+    .. [1] `Meteonorm
+       <https://meteonorm.com/>`_
+    .. [2] `Meteonorm API
+       <https://docs.meteonorm.com/docs/getting-started>`_
+    .. [3] `Meteonorm API reference
+       <https://docs.meteonorm.com/api>`_
+    """
+    params = {
```

Comment:
Seems like `_get_meteonorm` could be made flexible enough to accommodate the TMY endpoint as well, but okay with me to not pursue that in this PR.